packages/dashboard/src/components/VisualizationRow.tsx

```diff
@@ -99,108 +102,126 @@ const VisualizationRow: React.FC<VizRowProps> = ({ filteredDataOverride, row, ro
-                    config={visualizationConfig}
-                    isEditor={false}
-                    setConfig={newConfig => {
-                      updateChildConfig(col.widget, newConfig)
-                    }}
-                    isDashboard={true}
-                    configUrl={config.table && config.table.show && config.datasets && visualizationConfig.table && visualizationConfig.table.showDataTableLink ? tableLink : undefined}
-                  />
-                )}
-                {visualizationConfig.type === 'markup-include' && (
-                  <CdcMarkupInclude
-                    key={col.widget}
-                    config={visualizationConfig}
-                    configUrl={undefined}
-                    isDashboard={true}
-                    isEditor={false}
-                    setConfig={newConfig => {
-                      updateChildConfig(col.widget, newConfig)
-                    }}
-                  />
-                )}
-                {visualizationConfig.type === 'filtered-text' && (
-                  <CdcFilteredText
-                    key={col.widget}
-                    config={visualizationConfig}
-                    isEditor={false}
-                    setConfig={newConfig => {
-                      updateChildConfig(col.widget, newConfig)
-                    }}
-                    isDashboard={true}
-                    configUrl={undefined}
-                  />
-                )}
-                {visualizationConfig.type === 'filter-dropdowns' && !hideFilter && (
-                  <React.Fragment key={col.widget}>
-                    <Filters hide={visualizationConfig.hide} filters={config.dashboard.sharedFilters} apiFilterDropdowns={apiFilterDropdowns} handleOnChange={handleOnChange} />
-                    <GoButton autoLoad={visualizationConfig.autoLoad} />
-                  </React.Fragment>
-                )}
-                {visualizationConfig.type === 'table' && (
-                  <DataTableStandAlone
-                    key={col.widget}
-                    updateConfig={newConfig => {
-                      updateChildConfig(col.widget, newConfig)
-                    }}
-                    visualizationKey={col.widget}
-                    config={visualizationConfig as TableConfig}
-                    viewport={currentViewport}
-                  />
-                )}
-                {visualizationConfig.type === 'footnotes' && <FootnotesStandAlone key={col.widget} visualizationKey={col.widget} config={visualizationConfig} viewport={currentViewport} />}
-              </div>
+              {row.expandCollapseAllButtons ? (
+                <div className={`col-${col.width} ${!shouldShow ? 'd-none' : ''}`}>
+                  <div className='collapsable-multiviz-container'>
+                    <CollapsibleVisualizationRow allExpanded={allExpanded} fontSize={'26px'} groupName={groupName} currentViewport={currentViewport}>
+                      {body}

```

Comment:
this is more of a nit, but I really don't like the pattern of declaring a body and then conditionally placing it. To me that means you should have a separate wrapper component. Like I think if you created visualizationRowWrapper and made that the default export you could make the change in this file way smaller.

Reply:
I agree with you, I don't like the body pattern. I just didn't think of a better way since I was going to present this at 11:30. 
So I could pull body out and make a separate file and call it visualizationRowWrapper and then just pass all of the variables and functions that it might need.